### DISCUSSION TITLE
useComputed that also depends on props

### DISCUSSION THREAD

**Discussion by lorefnon** [preactjs/signals#485] [Q&A]
I understand that useComputed automatically tracks the signals accessed during its computation. 

So I assume that it is not safe to use useComputed if the computed value depends on some signals as well as some props  that the component received because it has no way of tracking the props. 

So, is falling back to useMemo the best option for such cases? Or should we consider a hook that auto-tracks accessed signals but also allows some external dependencies similar to useMemo (which need to be explicitly passed).

**Comment by rschristian** [comment]
Transferred to the signals repo as it's better situated here than Preact core.

Have you by chance skimmed https://github.com/preactjs/signals/issues/247? It might offer a solution for you, specifically https://github.com/preactjs/signals/issues/247#issuecomment-1301181739.

**Reply by lorefnon** [reply]
Ok, thank you. I didn't realize signals related discussions were in separate repo. 

So as I understand, signals is clearly a separate subsystem and my best bet for derived computation is to ensure that all the dependencies come through signals. Its not a big hassle though, can be dealt with easily.

**Reply by rschristian** [reply]
No worries!

They're different primatives for sure. You don't necessarily need to ensure that all bits of  data are signals, but it'll probably be easiest to at least massage the ones that aren't to be signals themselves.

It's not too different from standard (p)react there in that standard variables are a totally fine and reasonable way to use data in your typical web app, but to have an easier time with (p)react reactivity, putting or mirroring that into component state in some way tends to make things a bit easier.
